lib/ble.toit

```diff
@@ -278,13 +278,18 @@ class RemoteCharacteristic extends RemoteReadWriteElement_ implements Attribute:
 
   /**
   Writes the value of the characteristic on the remote device.
+
+  If $flush is true, waits until the data has been written. This is the default for
+    characteristics that require a response.
   */
-  write value/ByteArray -> none:
+  write value/ByteArray --flush/bool?=null -> none:
     if (properties & (CHARACTERISTIC-PROPERTY-WRITE
                       | CHARACTERISTIC-PROPERTY-WRITE-WITHOUT-RESPONSE)) == 0:
       throw "Characteristic does not support write"
 
-    write_ value --expects-response=(properties & CHARACTERISTIC-PROPERTY-WRITE) != 0
+    if not flush: flush = false
```

Comment:
This feels like a bad smell too. Why can't flush just have a proper default value (false)?